### PR TITLE
[oppo] Fix Play Mode and Disc Type updates

### PR DESCRIPTION
**Pull request by mlobstein**
fixes issue #12064 
In some configurations, the player sends unsolicited updates but the update message is an abbreviated code when compared to the longer response provided by the query command that must used in other configurations. The previous strategy to immediately call the query command upon receiving an unsolicited update was flawed in that the query command sometimes provided an outdated value. This PR fixes the issue for Play Mode and Disc Type by mapping the abbreviated unsolicited update code to the values provided by the query command. For Audio Type and Subtitle Type, a slight delay was added before calling the query command.

### CODE REVIEW

File: bundles/org.openhab.binding.oppo/src/main/java/org/openhab/binding/oppo/internal/handler/OppoHandler.java [hunk 1]
```diff
@@ -506,16 +514,17 @@ public void onNewMessageEvent(OppoMessageEvent evt) {
...
-                        // we got the disc type status update, throw it away
-                        // and call the query because the text output is better
-                        connector.sendCommand(OppoCommand.QUERY_DISC_TYPE);
                     case QDT:
-                        currentDiscType = updateData;
-                        updateChannelState(CHANNEL_DISC_TYPE, updateData);
+                        // try to normalize the slightly different responses between UDT and QDT
+                        final String discType = OppoStatusCodes.DISC_TYPE.get(updateData);
+                        currentDiscType = (discType != null ? discType : updateData);
+                        updateChannelState(CHANNEL_DISC_TYPE, (discType != null ? discType : updateData));
```
Comment (lolodomo) [inline bundles/org.openhab.binding.oppo/src/main/java/org/openhab/binding/oppo/internal/handler/OppoHandler.java:521]: Rather use currentDiscType.

File: bundles/org.openhab.binding.oppo/src/main/java/org/openhab/binding/oppo/internal/communication/OppoStatusCodes.java [hunk 2]
```diff
@@ -54,4 +56,36 @@
         ZOOM_MODE.put("11", "1/3");
         ZOOM_MODE.put("12", "1/4");
     }
+
+    // map to lookup disc type
+    public static final Map<String, String> DISC_TYPE = new HashMap<>();
+    static {
```
Comment (lolodomo) [inline bundles/org.openhab.binding.oppo/src/main/java/org/openhab/binding/oppo/internal/communication/OppoStatusCodes.java:62]: No chance to use Map.of? 
Maybe too many entries...

**Review by lolodomo — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.binding.oppo/src/main/java/org/openhab/binding/oppo/internal/handler/OppoHandler.java (hunk 1 above)
Comment (mlobstein) [reply bundles/org.openhab.binding.oppo/src/main/java/org/openhab/binding/oppo/internal/handler/OppoHandler.java]: yes, much simpler. fixed.

File: bundles/org.openhab.binding.oppo/src/main/java/org/openhab/binding/oppo/internal/communication/OppoStatusCodes.java (hunk 2 above)
Comment (mlobstein) [reply bundles/org.openhab.binding.oppo/src/main/java/org/openhab/binding/oppo/internal/communication/OppoStatusCodes.java:62]: Yes this constructor allows a max of 10 entries. If more than that, Map.ofEntries would have to be used.

Comment (lolodomo) [reply bundles/org.openhab.binding.oppo/src/main/java/org/openhab/binding/oppo/internal/communication/OppoStatusCodes.java:62]: So I let you change the code.

Comment (mlobstein) [reply bundles/org.openhab.binding.oppo/src/main/java/org/openhab/binding/oppo/internal/communication/OppoStatusCodes.java]: done

**Review by lolodomo — APPROVED** [review]
LGTM

**Pull request merged**